escposprinter/src/main/java/com/dantsu/escposprinter/connection/bluetooth/BluetoothPrintersConnections.java

```diff
@@ -33,6 +33,33 @@ public static BluetoothConnection selectFirstPaired() {
     }
 
 
+    /**
+     * Manual way to print by selected bluetooth printer
+     *
+     * @return a EscPosPrinterCommands instance
+     */
+    @Nullable
+    public static BluetoothConnection selectDevice(BluetoothConnection device) {
+        BluetoothPrintersConnections printers = new BluetoothPrintersConnections();
+        BluetoothConnection[] bluetoothPrinters = printers.getList();
+
+        if (bluetoothPrinters != null && bluetoothPrinters.length > 0) {
+            for (BluetoothConnection printer : bluetoothPrinters) {
+                try {
+                    if (device.getDevice().getAddress().equals(printer.getDevice().getAddress())){
+                        return printer.connect();
+                    }
+                } catch (EscPosConnectionException e) {
+                    e.printStackTrace();
+                }
+            }
+        }
+        return null;
+    }
```

Comment:
This method is useless. You do a while for finding and returning the same instance of parameter device. Just do a `device.connect()` in main activity.

Reply:
true, i just realize that i will remove it.